src/analyzer/controller/Controller.java

```diff
@@ -0,0 +1,42 @@
+package analyzer.controller;
+
+import analyzer.interactor.Analyzer;
+import analyzer.interactor.Finances;
+import analyzer.interactor.Report;
+import analyzer.interactor.RequestModel;
+import analyzer.presenters.html.HTMLPresenter;
+import analyzer.presenters.pdf.PDFPresenter;
+import com.sun.istack.internal.NotNull;
+
+/**
+ * Contains main logic, is invoked from main() method.
+ *
+ * @author Lizaveta Havina
+ */
+public class Controller {
+
+    private RequestModel requestModel;
+    private Analyzer analyzer;
+
+    public Controller(@NotNull final RequestModel requestModel,
+                      @NotNull final Analyzer analyzer) {
+        this.requestModel = requestModel;
+        this.analyzer = analyzer;
+    }
+
+    public void showData() {
+        Finances finances = requestModel.getFinances();
+        Report report = analyzer.analyze(finances);
+        Presenter presenter = null;
+        switch (requestModel.getType()) {
```

Comment:
presenter has to be an explicit dependency of the controller.

- There is no way to unit test controller when it calls presenter's methods
- Controller takes responsibility of creating a presenter

Reply:
Got it!